### PR TITLE
Update apache proton-j to 0.34.1 and fix flaky servicebus tests

### PR DESCRIPTION
**Pull request by lmolkova**

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -94,8 +94,7 @@ org.apache.maven:maven-core;3.5.4
 org.apache.maven.archetype:archetype-packaging;3.2.0
 org.apache.maven.plugins:maven-archetype-plugin;3.2.0
 org.apache.maven.plugins:maven-gpg-plugin;3.1.0
-org.apache.qpid:proton-j;0.33.8
-org.apache.qpid:qpid-jms-client;0.53.0
```
Comment (lmolkova) [inline eng/versioning/external_dependencies.txt:98]: not used anywhere

**Review by conniey — APPROVED** [review]
Lgtm

File: sdk/core/azure-core-amqp/pom.xml [hunk 2]
```diff
@@ -154,7 +154,7 @@
             <bannedDependencies>
               <includes>
                 <include>com.microsoft.azure:qpid-proton-j-extensions:[1.2.4]</include> <!-- {x-include-update;com.microsoft.azure:qpid-proton-j-extensions;external_dependency} -->
-                <include>org.apache.qpid:proton-j:[0.33.8]</include> <!-- {x-include-update;org.apache.qpid:proton-j;external_dependency} -->
+                <include>org.apache.qpid:proton-j:[0.34.1]</include> <!-- {x-include-update;org.apache.qpid:proton-j;external_dependency} -->
```
Comment (anuchandy) [inline sdk/core/azure-core-amqp/pom.xml:157]: It seems this proton-j-extensions also depends on proton-j version, but it's a library not in this mono repro, so we need to track down who owns it and how to release it.

https://github.com/Azure/qpid-proton-j-extensions/blob/1c372719aa1403f4ed2b18468462e07276cb128b/pom.xml#L48

**Review by anuchandy — APPROVED** [review]

File: sdk/core/azure-core-amqp/pom.xml (hunk 2 above)
Comment (anuchandy) [reply sdk/core/azure-core-amqp/pom.xml:157]: Okay direct owners of this library is JavaSDK team now. I can see Connie, me and v-team in the owner list, we can sync internally.

**Pull request merged**